packages/server/graphql/mutations/startCheckIn.ts

```diff
@@ -17,35 +17,62 @@ import createGcalEvent from './helpers/createGcalEvent'
 import createNewMeetingPhases from './helpers/createNewMeetingPhases'
 import isStartMeetingLocked from './helpers/isStartMeetingLocked'
 import {IntegrationNotifier} from './helpers/notifications/IntegrationNotifier'
+import maybeCreateOneOnOneTeam from './helpers/maybeCreateOneOnOneTeam'
+import CreateOneOnOneTeamInput, {
+  CreateOneOnOneTeamInputType
+} from '../public/types/CreateOneOnOneTeamInput'
 
 export default {
   type: new GraphQLNonNull(StartCheckInPayload),
   description: 'Start a new meeting',
   args: {
     teamId: {
-      type: new GraphQLNonNull(GraphQLID),
+      type: GraphQLID,
       description: 'The team starting the meeting'
     },
     gcalInput: {
       type: CreateGcalEventInput,
       description: 'The gcal event to create. If not provided, no event will be created'
+    },
+    oneOnOneTeamInput: {
+      type: CreateOneOnOneTeamInput,
+      description: 'One-on-One ad-hoc team to create. If provided, teamId ignored'
     }
   },
   async resolve(
     _source: unknown,
-    {teamId, gcalInput}: {teamId: string; gcalInput?: CreateGcalEventInputType},
-    {authToken, socketId: mutatorId, dataLoader}: GQLContext
+    {
+      teamId: existingTeamId,
+      gcalInput,
+      oneOnOneTeamInput
+    }: {
+      teamId: string
+      gcalInput?: CreateGcalEventInputType
+      oneOnOneTeamInput?: CreateOneOnOneTeamInputType
+    },
+    context: GQLContext
   ) {
     const r = await getRethink()
+    const {authToken, socketId: mutatorId, dataLoader} = context
     const operationId = dataLoader.share()
     const subOptions = {mutatorId, operationId}
     // AUTH
     const viewerId = getUserId(authToken)
-    if (!isTeamMember(authToken, teamId)) {
-      return standardError(new Error('Team not found'), {userId: viewerId})
+
+    let teamId = existingTeamId
```

Comment:
+1 I find the let a bit hard to follow. 

The first if check could check `existingTeamId` to help avoid `let`. 

Something like: 

```
    if (existingTeamId) {
      if (!isTeamMember(authToken, existingTeamId)) {
        return standardError(new Error('Team not found'), {userId: viewerId})
      }
      const unpaidError = await isStartMeetingLocked(existingTeamId, dataLoader)
      if (unpaidError) return standardError(new Error(unpaidError), {userId: viewerId})
    }
    const viewer = await dataLoader.get('users').loadNonNull(viewerId)
    const teamId = oneOnOneTeamInput ? await maybeCreateOneOnOneTeam(viewer, oneOnOneTeamInput, context) : existingTeamId
    if (!teamId) {
      return standardError(new Error('Must provide teamId or ad-hoc team'), {userId: viewerId})
    }
```

But that's just personal preference. Maybe you find that is harder to read.